src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHash128.cs

```diff
@@ -0,0 +1,460 @@
+
+            accLow = Avalanche(accLow);
+            accHigh = Avalanche(accHigh);
+
+            uint bound = ((length - (32 * 4)) / 32);
+            if (bound != 0)
+            {
+                Mix32Bytes(ref accLow, ref accHigh, source + (32 * 4), source + (32 * 4) + 16, DefaultSecret3UInt64_0, DefaultSecret3UInt64_1, DefaultSecret3UInt64_2, DefaultSecret3UInt64_3, seed);
+            }
+
+            if (bound >= 2)
+            {
+                Mix32Bytes(ref accLow, ref accHigh, source + (32 * 5), source + (32 * 5) + 16, DefaultSecret3UInt64_4, DefaultSecret3UInt64_5, DefaultSecret3UInt64_6, DefaultSecret3UInt64_7, seed);
+            }
+
+            if (bound == 3)
+            {
+                Mix32Bytes(ref accLow, ref accHigh, source + (32 * 6), source + (32 * 6) + 16, DefaultSecret3UInt64_8, DefaultSecret3UInt64_9, DefaultSecret3UInt64_10, DefaultSecret3UInt64_11, seed);
+            }
+
+            Mix32Bytes(ref accLow, ref accHigh, source + length - 16, source + length - 32, 0x4F0BC7C7BBDCF93F, 0x59B4CD4BE0518A1D, 0x7378D9C97E9FC831, 0xEBD33483ACC5EA64, 0 - seed);
+
+            return AvalancheHash(accLow, accHigh, length, seed);
+        }
+
+        private static Hash128 HashLengthOver240(byte* source, uint length, ulong seed)
+        {
+            Debug.Assert(length > 240);
+
+            fixed (byte* defaultSecret = &MemoryMarshal.GetReference(DefaultSecret))
+            {
+                byte* secret = defaultSecret;
+                if (seed != 0)
+                {
+                    byte* customSecret = stackalloc byte[SecretLengthBytes];
+                    DeriveSecretFromSeed(customSecret, seed);
+                    secret = customSecret;
+                }
+
+                ulong* accumulators = stackalloc ulong[AccumulatorCount];
+                InitializeAccumulators(accumulators);
+
+                HashInternalLoop(accumulators, source, length, secret);
+
+                Hash128 h128;
+                h128.Low64 = MergeAccumulators(accumulators, secret + SecretMergeAccsStartBytes, length * Prime64_1);
+                h128.High64 = MergeAccumulators(accumulators, secret + SecretLengthBytes - AccumulatorCount * sizeof(ulong) - SecretMergeAccsStartBytes, ~(length * Prime64_2));
+                return h128;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Hash128 AvalancheHash(ulong accLow, ulong accHigh, uint length, ulong seed)
+        {
+            ulong h128Low = accLow + accHigh;
+            ulong h128High = (accLow * Prime64_1)
+                          + (accHigh * Prime64_4)
+                          + ((length - seed) * Prime64_2);
+            h128Low = Avalanche(h128Low);
+            h128High = 0ul - Avalanche(h128High);
+            return new Hash128(h128Low, h128High);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ulong XorShift64(ulong v64, int shift)
```

Comment:
There's an XorShift with the same signature in XxHashShared.  Can they be deduped?

Reply:
Fixed by commit af4cb41